### PR TITLE
Reszie vcpu ram

### PR DESCRIPTION
**Pull request by zhang-shengping**

### CODE REVIEW

File: bigip/resize.sh [hunk 1]
```diff
@@ -0,0 +1,132 @@
+#/bin/bash -e
```
Comment (zhaoqin-github) [inline bigip/resize.sh:1]: Please rename it to reseize_bigip.sh. Its name will be consistent with other commands.

File: bigip/resize.sh [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+source common.sh
+
+reset_signals
+
+usage() {
```
Comment (zhaoqin-github) [inline bigip/resize.sh:7]: Does your command only support to resize one BIG-IP VE? Other commands support batch operation to create/activate a set of BIG-IP at the same time. Is it possible to make this command support batch operation?

File: bigip/resize.sh [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+# resize cpu
+if [[ ! -z $VCPU ]]; then
+  resize_vcpu $QEMU_CONN $BIGIP_NAME $VCPU
+fi
+# resize ram
+if [[ ! -z $RAM ]]; then
+  resize_ram $QEMU_CONN $BIGIP_NAME $RAM
+fi
+
+# start VM
```
Comment (zhaoqin-github) [inline bigip/resize.sh:128]: How long will it take to accomplish a resizing?

**Comment by zhaoqin-github** [comment]
> I think your commit history has some problem. Please fix it first. I need to look into the code for a little while..

I know the problem now.. Your branch is created from a wrong repo...

File: bigip/common.sh [hunk 4]
```diff
@@ -497,3 +497,169 @@ function replace_str(){
...
+# shutdown a VM gracefully
+# $1: QEMU connection
+# $2: vm name
+function shutdown(){
+  local host=$1 
+  local vm=$2
+  local state=$(virsh -c $host list | grep $vm | awk -F' ' '{print $3}')
+  
+  if [[ -n $state ]]; then
+    virsh -c $host shutdown $vm
```
Comment (zhaoqin-github) [inline bigip/common.sh:528]: If we really need to shutdown bigip by this command, we may use tmsh or icontrol to shutdown it. Not sure KVM command can gracefully shutdown a bigip.

File: bigip/resize.sh (hunk 1 above)
Comment (zhang-shengping) [reply bigip/resize.sh:1]: > Please rename it to reseize_bigip.sh. Its name will be consistent with other commands.

Ok, I will change it.

File: bigip/resize.sh (hunk 2 above)
Comment (zhang-shengping) [reply bigip/resize.sh:7]: > Does your command only support to resize one BIG-IP VE? Other commands support batch operation to create/activate a set of BIG-IP at the same time. Is it possible to make this command support batch operation?

resize is a continuous action, which combines several states and time-consume actions. I do not know it is good to do it as a batch. if an operator repeatedly running commands while the VM is still resizing. Well, if you want it batch, I can change it batch.

File: bigip/resize.sh (hunk 3 above)
Comment (zhang-shengping) [reply bigip/resize.sh:128]: > How long will it take to accomplish a resizing?

[root@localhost bigip]# time ./resize_bigip.sh -v8 -r20G --name  Peter-3-10.250.100.103
shutdown vm Peter-3-10.250.100.103
Domain Peter-3-10.250.100.103 is being shutdown

wait for host 10.250.100.103 shutdown ...
set 8 vcpus in vm Peter-3-10.250.100.103 on host qemu+ssh:///system


set 20G ram in vm Peter-3-10.250.100.103 on host qemu+ssh:///system


Domain Peter-3-10.250.100.103 started

wait for host 10.250.100.103 startup ...
finsih resize

real	1m49.094s
user	0m12.686s
sys	0m13.474s
If everything goes well, It takes about 50 seconds.

File: bigip/common.sh (hunk 4 above)
Comment (zhang-shengping) [reply bigip/common.sh:528]: > If we really need to shutdown bigip by this command, we may use tmsh or icontrol to shutdown it. Not sure KVM command can gracefully shutdown a bigip.

```
       shutdown domain [--mode MODE-LIST]
           Gracefully shuts down a domain.  This coordinates with the domain
           OS to perform graceful shutdown, so there is no guarantee that it
           will succeed, and may take a variable length of time depending on
           what services must be shutdown in the domain.
```
According to the MAN of virsh and my observation of VNC log, it seems shutdown step by step, for example, httpd service shutdown, ssh shutdown, etc shutdown... I could put this in regression test task for further verification.

File: bigip/resize.sh (hunk 3 above)
Comment (zhaoqin-github) [reply bigip/resize.sh:128]: > > How long will it take to accomplish a resizing?
> 
> ```
> [root@localhost bigip]# time ./resize_bigip.sh -v8 -r20G --name  Peter-3-10.250.100.103
> shutdown vm Peter-3-10.250.100.103
> Domain Peter-3-10.250.100.103 is being shutdown
> 
> wait for host 10.250.100.103 shutdown ...
> set 8 vcpus in vm Peter-3-10.250.100.103 on host qemu+ssh:///system
> 
> 
> set 20G ram in vm Peter-3-10.250.100.103 on host qemu+ssh:///system
> 
> 
> Domain Peter-3-10.250.100.103 started
> 
> wait for host 10.250.100.103 startup ...
> finsih resize
> 
> real	1m49.094s
> user	0m12.686s
> sys	0m13.474s
> ```
> 
> If everything goes well, It takes about 50 seconds.

109 seconds?

File: bigip/common.sh [hunk 5]
```diff
@@ -497,3 +497,169 @@ function replace_str(){
...
+  local QEMU_CONN
+  # var HOST not need to be treat as single word as "$HOST"
+  if [[ ! -z $host ]]; then
+    if ! validate_ip $host ; then
+      echo >&2 "\"$host\" is not an IPv4 address"
+      exit 1
+    fi
+    QEMU_CONN=qemu+ssh://$host/system
+  else 
+    QEMU_CONN=qemu+ssh:///system
```
Comment (zhaoqin-github) [inline bigip/common.sh:514]: Needn't use ssh for local kvm. Otherwise, you also need to authorized your certificate in your local ssh.

**Pull request closed without merging**